codice/ddf
# [2.19.x] DDF-UI-146 Add error messaging for when bounding box south >= north

#### What does this PR do?
This PR adds inline and pop-up error messaging for when a bounding box's south value is greater than or equal to the north value. It prevents the invalid bounding box from rendering on both maps and prevents the user from executing the search until the error is resolved.
This PR also refactors some of the validation functions in dms-utils and dd-utils
#### Who is reviewing it? 
@andrewzimmer @cassandrabailey293 @zta6 
#### Select relevant component teams: 
@codice/ui 
#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@shaundmorris 
#### How should this be tested?
- Create an anyGeo search with a bounding box 
- Enter south values that are less than or equal to a north value
- Verify that you see the inline error message (first screenshot below)
- Switch to different coordinate systems and verify the message persists (USNG and UTM messages will be worded differently; see second screenshot below)
- Verify that nothing is rendered on the map (both 2D and 3D)
- Attempt to execute the search and verify that you see the generic "bounding box is invalid" pop-up error message
- Verify there is no regression with previous inline & pop-up error messaging related to bounding box geo searches
#### Any background context you want to provide?
#### What are the relevant tickets?
Fixes: #codice/ddf-ui#146
G-7959
#### Screenshots
![image](https://user-images.githubusercontent.com/39737329/78170784-b6006780-7410-11ea-8df6-9e3fa4513ee1.png)
![image](https://user-images.githubusercontent.com/39737329/78170861-d4fef980-7410-11ea-9186-d76f98dfa764.png)
#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
